MultiplayerMod/Platform/Steam/Network/Messaging/Surrogates/MinionStartingStatsSurrogate.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+
+namespace MultiplayerMod.Platform.Steam.Network.Messaging.Surrogates;
+
+public class MinionStartingStatsSurrogate : ISerializationSurrogate, ISurrogateType {
+
+    public Type Type => typeof(MinionStartingStats);
+
+    public void GetObjectData(object obj, SerializationInfo info, StreamingContext context) {
+        var node = (MinionStartingStats) obj;
+        info.AddValue("Name", node.Name);
+        info.AddValue("NameStringKey", node.NameStringKey);
+        info.AddValue("GenderStringKey", node.GenderStringKey);
+        info.AddValue("stressTrait", node.stressTrait?.Id);
+        info.AddValue("joyTrait", node.joyTrait?.Id);
+        info.AddValue("congenitaltrait", node.congenitaltrait?.Id);
+        info.AddValue("stickerType", node.stickerType);
+        info.AddValue("voiceIdx", node.voiceIdx);
+        info.AddValue("personality", node.personality?.Id);
+        info.AddValue("Traits", node.Traits.Select(a => a?.Id).ToArray());
+        info.AddValue("StartingLevels.Keys", node.StartingLevels.Keys.ToArray());
```

Comment:
What type is `node.StartingLevels`? Can't we serialize it as a raw dictionary🤔?

Reply:
Nope. Dictionary won't get serialised. It seems like here we need to add primitive types only

Reply:
Wow, okay 🫤